chrysom/basicClient.go

```diff
@@ -85,22 +88,46 @@ const (
 // Items is a slice of model.Item(s) .
 type Items []model.Item
 
+type HTTPClient interface {
+	Do(*http.Request) (*http.Response, error)
+}
+
+type BasicClientIn struct {
+	fx.In
+
+	BasicClientConfig BasicClientConfig
+	// (Optional) Defaults to http.DefaultClient.
+	HTTPClient HTTPClient
```

Comment:
This allows custom http clients since servers like tr1d1um need it. ATM, the current setup forces the user to do some not so great things to accomplish the same thing.

Reply:
I don't follow why having the configuration for this feature in the BasicClientConfig is worse than here.  This to me makes configuring the right http client harder.

Reply:
`Harder` was too strong of a word.

Having the http client in BasicClientConfig will require the user to unmarshal their `BasicClientConfig` and then open it up and then add their own http client. I didn't like this because it requires us to modify `BasicClientConfig` struct during dependency injection.

I thought it would be cleaner to have the user inject their own http.client with fx and have ancla use that.